win32ss/user/ntuser/security.c

```diff
@@ -196,7 +196,7 @@ IntQueryUserSecurityIdentification(
                                      NULL,
                                      0,
                                      &BufferLength);
-    if (!NT_SUCCESS(Status) && Status == STATUS_BUFFER_TOO_SMALL)
+    if (Status == STATUS_BUFFER_TOO_SMALL)
```

Comment:
at line 214 (the end of this if-block), you may add a
```c
else if (!NT_SUCCESS(Status))
{
    //some err...
    ZwClose(Token);
    return Status;
}
```
in case that first ZwQueryInformationToken fails but for a reason diffferent from buffer too small.